### PR TITLE
More PropertyGrid related cleanup.

### PR DESCRIPTION
**Pull request by JeremyKuhne**
- Make more methods that should have been properties, properties.
- More devirtualization, sealing, and access modifier restriction.
- Move more collections to generics for clarity
- Move outline rectangle reset into GridEntry.
- Start on helper class for TypeDescriptor.
- Remove BinaryFormatter from MergePropertyDescriptor.
- Fix issue in SingleSelectRootGridEntry where Children would potentially be created for disposal.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/5777)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/SingleSelectRootGridEntry.cs [hunk 1]
```diff
@@ -97,7 +96,7 @@ public override AttributeCollection BrowsableAttributes
 
                 _browsableAttributes = value;
 
-                if (!same && Children is not null && Children.Count > 0)
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/SingleSelectRootGridEntry.cs:100]: Hitting the `Children` property creates the children. `ChildCount` specifically avoids this. Don't want to create children just to dispose of them.

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/MergePropertyDescriptor.cs [hunk 2]
```diff
@@ -133,48 +125,51 @@ private object CopyValue(object value)
...
-
-                // If that didn't work, try conversion to/from string
-                if (clonedValue is null && converter.CanConvertTo(typeof(string)) && converter.CanConvertFrom(typeof(string)))
-                {
-                    object stringRep = converter.ConvertToInvariantString(value);
-                    clonedValue = converter.ConvertFromInvariantString((string)stringRep);
-                }
             }
 
-            // How about serialization?
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/MergePropertyDescriptor.cs:160]: As we need to get rid of the `BinaryFormatter` and this was a last ditch effort I think we're pretty safe to just remove it. Workarounds, should they be necessary, would be to implement `ICloneable` or implement a string converter.

**Comment by JeremyKuhne** [comment]
Arm builds aren't finding the Arcade SDK.

```
D:\workspace\_work\1\s\artifacts\toolset\restore.proj : error MSB4236: The SDK 'Microsoft.DotNet.Arcade.Sdk' specified could not be found.
```

**Comment by RussKie** [comment]
Raised https://teams.microsoft.com/l/message/19:afba3d1545dd45d7b79f34c1821f6055@thread.skype/1631750375573?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=4d73664c-9f2f-450d-82a5-c2f02756606d&parentMessageId=1631750375573&teamName=.NET%20Core%20Eng%20Services%20Partners&channelName=First%20Responders&createdTime=1631750375573

**Review by RussKie — APPROVED** [review]

**Pull request merged**